src/test/ui/const-eval/simple_with_undef.rs

```diff
@@ -0,0 +1,16 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// compile-pass
+
+const PARSE_BOOL: Option<&'static str> = None;
+static FOO: (Option<&str>, u32) = (PARSE_BOOL, 42);
```

Comment:
Where is there an undef here? The enum gets layout optimized so writing `None` initializes the entire thing, does it not?

Reply:
I had a regression when building stage 2 that didn't occur in the test suite, so I added a test. The issue was that some function didn't read the discriminant and figure out the variant layout before reading fields, so it was reading fields by enum indices (where 0 is the discriminant).

Reply:
I agree with the test, I am just confused by the filename, which says that `undef` is relevant here.